File: java/src/Block/DropIndexBlock.java
/**
 * This file copyright (c) 2015-2016, Jing Fan, Adalbert Gerald Soosai Raj, and Jignesh M. Patel
 *
 * See the file CREDITS.txt in the root directory for details.
 **/

package Block;

/**
 * @brief The SQL block for dropping an index.
 */
public class DropIndexBlock extends Block {

  /**
   * @brief Constructor.
   * @param stage The string indicate the stage of this code block.
   * @param indent The indent level.
   * @param index The name of the index to be dropped.
   * @param tbName The name of the table containing the index.
   */
  public DropIndexBlock(String stage, int indent, String index, String tbName) {
    super(stage, indent);
    this.append("DROP INDEX " + index + " ON " + tbName + ";");
    this.sql = this.sb.toString();
  }

}

File: python/src/Blocks/DropIndexBlock.py
from Block import Block

class DropIndexBlock(Block):
    '''
    generates sql to drop an index on a table
    '''
    
    def __init__(self,stage,indent,index,tbName):
        '''
        constructor: 
        @param stage: stage of the block 
        @type stage: string
        @param indent: indent level
        @type indent: int
        @param index: index to drop
        @type index: string
        @param tbName: table name 
        @type tbName: string
        '''
        super(DropIndexBlock,self).__init__(stage,indent)
        self.append("DROP INDEX IF EXISTS " + str(index) + ";")
        self.sql=self.sb